contrib/scoverage/api/src/mill/contrib/scoverage/api/ScoverageReportWorkerApi.scala

```diff
@@ -1,35 +1,19 @@
 package mill.contrib.scoverage.api
 
 import mill.api.Ctx
+import unroll.Unroll
 
 trait ScoverageReportWorkerApi {
   import ScoverageReportWorkerApi._
 
-  @deprecated("Use other overload instead.", "Mill after 0.10.7")
-  def report(
-      reportType: ReportType,
-      sources: Seq[os.Path],
-      dataDirs: Seq[os.Path]
-  )(implicit
-      ctx: Ctx
-  ): Unit = {
-    report(reportType, sources, dataDirs, ctx.workspace)
-  }
-
   def report(
       reportType: ReportType,
       sources: Seq[os.Path],
       dataDirs: Seq[os.Path],
-      sourceRoot: os.Path
+      @Unroll sourceRoot: os.Path = null
```

Comment:
Can you elaborate how the use of the unroll plugin changes the code? Will API users who freshly compile against the binary code, always use the new API? Are the backward-compatible stubs only for binary compat, or are they still public non-deprecated API?

Reply:
@lefou it generates bincompat forwarder overloads. The plugin repo has some description and links to further discussion. I hope to get this into the Scala language via a SIP

https://github.com/com-lihaoyi/unroll

For now, the generated forwarders aren't hidden. It doesn't really matter, since it's just a forwarder, calling it does the same thing as calling the "real" definition. There's some discussion on how to hide them, but nothing really decided upon

Reply:
Understood. Would be nice if there is an option to mark the forwarders as deprecated, since then, it's in the hand of the API user to hack around them and produce bytecode that may be more safe, if needed. But it's not needed here, I think.